source/module.c

```diff
@@ -237,6 +298,105 @@ int aws_napi_callback_clean_up(struct aws_napi_callback *cb) {
     return AWS_OP_SUCCESS;
 }
 
+static void s_handle_failed_callback(napi_env env, napi_status status) {
+    /* Figure out if there's an exception pending, if so, no callbacks will ever succeed again until it's cleared */
+    bool pending_exception = status == napi_pending_exception;
+    if ((status = napi_is_exception_pending(env, &pending_exception))) {
```

Comment:
it looks like you're setting an initial value for `pending_exception`, then passing it as an out-param?
is that on purpose?

Reply:
So, if the status is napi_pending_exception, there is DEFINITELY an exception. However, the other statuses could ALSO mean there is an exception, which is why you have to ask. It doesn't really matter, unless ```napi_is_exception_pending()``` fails somehow. It's just the default value, I could have easily used ```false``` or ```true```.